### ISSUE TITLE
Bug: SA1000 handles single line (un)checked blocks incorrectly

### ISSUE THREAD

**Issue by vweijsters** [DotNetAnalyzers/StyleCopAnalyzers#563]
While testing some code I got an SA1000 diagnostic reported for:

```C#
checked { i = i + 1; }
```
which is not reported by the original StyleCop.
The line of code above should only report SA1501.

I think we should modify the current SA1000 implementation to support 2 usage scenarios.
```C#
// This should report SA1000
i = checked (i+1)

// This should not report SA1000
checked { i = i + 1 }
```

This would also mean adapting the code fix.

`unchecked` should be handled in the same way as well.

**Comment by sharwell** [comment]
Makes sense to me.

**Comment by pdelvo** [comment]
grabbed
